cpp/ql/src/semmle/code/cpp/ir/dataflow/DefaultTaintTracking.qll

```diff
@@ -139,10 +149,10 @@ private predicate writesVariable(StoreInstruction store, Variable var) {
 // previously suppressed by this predicate by coincidence.
 private predicate hasUpperBoundsCheck(Variable var) {
   exists(RelationalOperation oper, VariableAccess access |
```

Comment:
Would you go so far as to change `RelationalOperation` to `ComparisonOperation`? I think we've seen some code recently that took a size from the network, verified that it was equal to some constant, and then allocated that size of memory.

It doesn't have to be in this PR.

Reply:
I'll look into this as a follow-up (I think I remember seeing that code).